### PR TITLE
ICD builder refactoring

### LINKED ISSUE #178: Tools > "Generate ICD" - missing fieldPrefix AUTO support
Currently options used during ICD generation are hard-coded, should be configurable by user.

### LINKED ISSUE #181: Display ICD after generating it

### PR DESCRIPTION
**Pull request by hcorg**
Fixes #181 
Fixes #178

### CODE REVIEW

File: src/icd/icdbuilder.cpp [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+std::shared_ptr<BuildConfiguration> prepareClone(BuildConfiguration *buildConfig)
+{
+    auto result = std::shared_ptr<BuildConfiguration>{buildConfig};
+    QObject::connect(BuildManager::instance(),
+                     &BuildManager::buildQueueFinished,
+                     [result](bool) mutable { result.reset(); });
+    return result;
+}
+
+std::shared_ptr<BuildConfiguration> config(Project *project)
```
Comment (Lurkerpas) [inline src/icd/icdbuilder.cpp]: I am not sure about the convention used in this project, but I'm more used to functions containing verbs, like getConfig

File: src/icd/icdbuilder.cpp [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+
+IcdBuilder::IcdBuilder() {}
+IcdBuilder::~IcdBuilder() {}
+
+void IcdBuilder::run(ProjectExplorer::Project *project)
+{
+    auto steps = buildSteps(project);
+    auto step = (steps != nullptr) ? modifiedBuildStep(steps) : nullptr;
+
+    if (!step) {
```
Comment (Lurkerpas) [inline src/icd/icdbuilder.cpp]: I believe != nullptr is the preferred idiom

File: src/icd/icdbuilder.cpp [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+
+BuildStep *modifiedBuildStep(BuildStepList *steps)
+{
+    auto makeStep = modifiedMakeBuildStep(steps);
+    if (makeStep != nullptr)
+        return makeStep;
+    return modifiedCMakeBuildStep(steps);
+}
+
+std::shared_ptr<BuildConfiguration> prepareClone(BuildConfiguration *buildConfig)
```
Comment (Lurkerpas) [inline src/icd/icdbuilder.cpp]: Please change name to something more descriptive (maybe createSuicidalClone :P)

Comment (Lurkerpas) [reply src/icd/icdbuilder.cpp]: createAndConnectASuicidalClone ^_^

File: src/icd/icdbuilder.cpp (hunk 1 above)
Comment (hcorg) [reply src/icd/icdbuilder.cpp]: although it is a Qt-style to name getters using nouns, it's not really a getter, but builder, I've renamed it accordingly

File: src/icd/icdbuilder.cpp (hunk 2 above)
Comment (hcorg) [reply src/icd/icdbuilder.cpp]: too many years using smart ptrs without C++11 :) thanks, fixed

File: src/icd/icdbuilder.cpp (hunk 3 above)
Comment (hcorg) [reply src/icd/icdbuilder.cpp]: `wrapForProperDeletion(factory->clone(target, buildConfig));`

**Pull request merged**